### ISSUE TITLE
Create target directory if missing

### ISSUE THREAD

**Issue by cvetanov** [havardh/javaflow-maven-plugin#4]
Hello,

While using the plugin, we ran into an issue where the target directory (custom one, not the default /target) was not existing so we got an `NoSuchFileException`. Creating the folder manually fixed this and we do not have this issue anymore.

However, I was thinking, what if there is an option which will tell the plugin to create the target folder (if missing)? Something like `createIfMissing`, or feel free to propose a better name.
What do you think about it?

I could submit a PR for that if that's fine with you.

Regards,
Goce

**Comment by havardh** [comment]
Hi @cvetanov.

Thank you for submitting this issue and I would absolutely welcome a PR. :D

I'm not sure that we would need to put this functionality behind a flag. I think we should create the folder if it is missing by default. Do you know how this is handled by other maven plugins? Is it common to require a flag for this functionality?

**Comment by cvetanov** [comment]
Hi,

You are right, we should support this by default, no need for a flag.
I do not have any experience of how this is handled in other plugins, but I can do some research.
At the moment, all I can think of is to do it in the plugin implementation (by using [`mkdir()`](https://docs.oracle.com/javase/7/docs/api/java/io/File.html#mkdir())), but before jumping to it, I will for sure do some research on this topic.

If any other users have this problem until we come up with the feature in the plugin itself, I can share our solution.

We are using the `maven-antrun-plugin` right **before** the `javaflow-maven-plugin` and using the `mkdir` task (which creates a folder or does nothing if the folder is already existing).

```
<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-antrun-plugin</artifactId>
  <executions>
    <execution>
      <id>create targetDirectory for javaflow plugin</id>
      <phase>generate-resources</phase>
      <configuration>
        <target description="creates the targetDirectory for javaflow plugin (if not existing)">
          <mkdir dir="${project.basedir}/customTargetDirectory"/>
        </target>
      </configuration>
      <goals>
        <goal>run</goal>
      </goals>
    </execution>
  </executions>
</plugin>
```

Will post a follow up as soon as I have any updates.

Regards,
Goce

**Comment by havardh** [comment]
Great, thanks for sharing your workaround. :+1: Please do not feel obligated to do research on plugins, do it if you feel like it. I propose we do it without the flag (if you do not have compelling arguments against it).

Are you familiar with forking and creating PRs on GitHub? If not this is a good resource for getting started http://makeapullrequest.com.
 I do follow the [angular commit convention](https://gist.github.com/stephenparish/9941e89d80e2bc58a153) for this repository and it would be nice if you could make your commits follow it as well. Take a look at the git history for examples.

**Comment by havardh** [comment]
You might consider using [`Files.createDirectories`](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html) to have it create missing parent directories too. :)

**Comment by havardh** [comment]
Resolved by #5

**Comment by havardh** [comment]
Released in v1.3.0. @cvetanov let me know if this release has solved your issue :)
